### PR TITLE
Sensor and actuator pages

### PR DESCRIPTION
**Pull request by fabienrohrer**
Preview:

- https://www.cyberbotics.com/doc/guide/actuators?version=enhancement-sensor-and-actuator-pages
- https://www.cyberbotics.com/doc/guide/sensors?version=enhancement-sensor-and-actuator-pages

---

- [x] List of generic sensors (+ icons, description and link the reference manual)
- [x] List of generic actuators (+ icons, description and link the reference manual)
- [x] Migrate the sensors described in the automobile book to these new pages (one page per category)

### CODE REVIEW

**Comment by fabienrohrer** [comment]
I'm hesitating to "open" the branded sensor list in this PR or to already merge this as-is (which is already a good milestone).

**Comment by omichel** [comment]
I believe we can already merge this PR with only the generic sensors and actuators.
We could add the "branded" devices in another PR, but I am not sure "branded" is a good term to describe them.

**Comment by fabienrohrer** [comment]
@tn12787 Do you have a better idea to name this section?

**Comment by fabienrohrer** [comment]
Note: This PR aims to have a good list of sensors and actuators and so to migrate the sensors described in the automobile book in the user guide as-is.

The following topics will be addressed in other PRs:

- The list is not exhaustive: remains to document: Hokuyo, Kinect and Servo.
- The icons will be regenerated later (this is related with the David work). The icons names may change in this operation.
- The automobile book should be entirely migrated into the guide.

**Comment by fabienrohrer** [comment]
👍

**Comment by omichel** [comment]
:eyes:

File: guide/actuators.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+### Generic Actuator List
+
+| Icon | Device | Description |
+| :---: | --- | --- |
+| ![Brake.png](images/actuators/Brake.png) | [Brake](../reference/brake.md) | *Simulates a mechanical braking system.* |
+| ![Connector.png](images/actuators/Connector.png) | [Connector](../reference/connector.md) | *Simulates a mechanical and breakable docking system.* |
+| ![Display.png](images/actuators/Display.png) | [Display](../reference/display.md) | *Can be used to paint to a 2D image using low-level API functions which can be rendered into a 2D overlay and/or into a texture displayed in the 3D world.* |
+| ![Emitter.png](images/actuators/Emitter.png) | [Emitter](../reference/emitter.md) | *Simulates radio, serial or infra-red emitters in order to send data to other robots.* |
+| ![LinearMotor.png](images/actuators/LinearMotor.png) | [LinearMotor](../reference/linearmotor.md) | *Simulates a motor constrained on an axis.* |
```
Comment (omichel) [inline guide/actuators.md]: I would add "linear" here.

File: guide/actuators.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+| ![LED.png](images/actuators/LED.png) | [LED](../reference/led.md) | *Simulates a light emitting diode (LED).* |
+| ![Muscle.png](images/actuators/Muscle.png) | [Muscle](../reference/muscle.md) | *Can be added to a motor to display an artificial muscle.* |
```
Comment (omichel) [inline guide/actuators.md]: I would rephrase it as "Simulates the shape of an artificial muscle that can be added to a motor".

File: guide/actuators.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+| ![Pen.png](images/actuators/Pen.png) | [Pen](../reference/pen.md) | *Simulates a pen able to draw on every graphical geometric primitive.* |
```
Comment (omichel) [inline guide/actuators.md]: every &rarr; any

File: guide/actuators.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+| ![Propeller.png](images/actuators/Propeller.png) | [Propeller](../reference/propeller.md) | *Simulates a marine or aircraft propeller.* |
+| ![RotationalMotor.png](images/actuators/RotationalMotor.png) | [RotationalMotor](../reference/rotationalmotor.md) | *Simulates a rotational motor.* |
+| ![Speaker.png](images/actuators/Speaker.png) | [Speaker](../reference/speaker.md) | *Simulates a loudspeaker device.* |
+| ![Track.png](images/actuators/Track.png) | [Track](../reference/track.md) | *Simulates tracks for conveyor belts or tank robots.* |
```
Comment (omichel) [inline guide/actuators.md]: I would use the singular form as in other cases: "Simulates a track for..."

File: guide/sensors.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+## Sensors
+
+Webots can simulate a lot of standard robotic sensors.
+Using the nodes described [below](#generic-sensor-list) and their corresponding API is sufficient in most of the cases.
+
+More specific sensors are built on the top of these generic nodes, thanks to the [PROTO system](../reference/proto.md).
+These sensors are listed by category [in the section below](#branded-sensors).
```
Comment (omichel) [inline guide/sensors.md]: The (#branded-sensors) link is wrong. It wasn't detected by our test system?

Comment (fabienrohrer) [reply guide/sensors.md]: Strange. I'm taking a look at this.

Comment (fabienrohrer) [reply guide/sensors.md]: Fixed in: https://github.com/omichel/webots-doc/pull/737

**Comment by fabienrohrer** [comment]
jackpot :-) I fixed 3 obvious bad access to memory thanks to these assertions.

**Comment by fabienrohrer** [comment]
👍
